src/gather.jl

```diff
@@ -0,0 +1,78 @@
+export gather, gather!
+
+"""
+    gather!(dst, src, idx, dims)
+
+Reverse operation of scatter, which gather data in `src` to destination according to `idx`.
+For each index `k` in `idx`, assign values to `dst` according to
+
+    dst[k...] = src[idx[k]...]
+
+# Arguments
+- `dst`: the destination where data would be assigned to.
+- `src`: the source data to be assigned.
+- `idx`: the mapping for assignment from source to destination.
+- `dims`: the number of dimensions to be view as an operational unit from the beginning.
+If dims=0 is given, operational unit is scalar. Default dims is 1.
+
+Examples for dims are lists here:
+
+- `dims=0`: dst[k...] = src[idx[k]...]
+- `dims=1`: dst[:, k...] .= src[:, idx[k]...]
+- `dims=2`: dst[:, :, k...] .= src[:, :, idx[k]...]
```

Comment:
do we have to pass a `dims` argument? can't the proper indexing be inferred from the dimensions of `dst` and `src` and from the eltype of `idx`?

Reply:
I implemented my own suggestion and removed `dims` :)